haystack/components/agents/agent.py

```diff
@@ -225,6 +225,10 @@ def __init__(
             component.set_input_type(self, name=param, type=config["type"], default=None)
         component.set_output_types(self, **output_types)
 
+        # Validate tools' state mappings against state schema
+        if tools:
+            self._validate_tools_against_state_schema(tools)
+
         self.tool_invoker_kwargs = tool_invoker_kwargs
         self._tool_invoker = None
         if self.tools:
```

Comment:
We already have a `self.tools` if check on line 234. Could we move this validation step in there?

Also does this validation work when using MCPToolset?